src/Module/NoScrape.php

```diff
@@ -22,12 +22,17 @@ public static function rawContent(array $parameters = [])
 	{
 		$a = DI::app();
 
-		$which = DI::args()->get(1);
-
-		$profile = 0;
-		if ((local_user()) && (DI::args()->get(2) === 'view')) {
+		if (isset($parameters['nick'])) {
+			// Get infos about a specific nick (public)
+			$which = $parameters['nick'];
+			$profile = 0;
+		} elseif (local_user() && isset($parameters['profile']) && DI::args()->get(2) == 'view') {
+			// view infos about a known profile (needs a login)
 			$which   = $a->user['nickname'];
-			$profile = DI::args()->get(1);
+			$profile = $parameters['profile'];
+		} else {
+			System::jsonError(404, 'Invalid request');
```

Comment:
This will not happen as the routes won't be triggered for anything other than the two cases above.

Reply:
It happens:
```console
localhost:~$ curl http://192.168.22.10/noscrape/friendica1/view
"Invalid request"
```
- valid route `/noscrape/{profile}/view`
- but not logged in

Reply:
Oooh, this should be a error number 401 or a 403 in the `/noscrape/{profile}/view` case instead then, with a "Authentication required" or "Forbidden" or "You must be logged in" message.